.github/workflows/image-cleanup.yml

```diff
@@ -0,0 +1,82 @@
+name: Container image cleanup
+
+on:
+# FIXME(chrisgacsal): re-enable PR and scheduled trigger after successful testing
+#  pull_request:
+#    types:
+#      - closed
+#  schedule:
+#    # At 06:00 on every day-of-week from Monday through Friday.
+#    # https://crontab.guru/#0_6_*_*_1-5
+#    - cron: '0 6 * * 1-5'
+  workflow_dispatch:
+    inputs:
+      cut-off:
+        required: false
+        type: string
+        description: |
+          The timezone-aware datetime you want to delete container versions that are older than.
+          The parsed datetime must contain a timezone.
+
+          The `dateparser` is ued to parse the cut-off specified. See: [dateparser](https://dateparser.readthedocs.io/en/latest/)
+        default: '14 days ago UTC'
+      dry-run:
+        required: false
+        type: boolean
+        description: Prints output showing images which would be deleted but does not actually delete any images.
+        default: true
+
+permissions:
+  packages: write
+
+jobs:
+  pull-request:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    name: Cleanup container images for Pull Request
+    steps:
+      - name: Remove images for PR#${{ github.event.pull_request.number }}
+        uses: snok/container-retention-policy@v2
+        with:
+          image-names: vmclarity-apiserver-dev,vmclarity-cli-dev,vmclarity-ui-backend-dev,vmclarity-ui-dev
```

Comment:
it might be a good idea to make this a top level workflow environment variable to avoid the duplication among all the jobs wdyt?

Reply:
Agreed. Fixed.